packages/host/app/components/operator-mode/code-mode.gts

```diff
@@ -261,10 +261,13 @@ export default class CodeMode extends Component<Signature> {
 
     return file(context, () => ({
       url: this.codePath!.href,
-      onStateChange: (state) => {
+      onStateChange: (state, newStateObj) => {
         if (state === 'not-found') {
           this.loadFileError = 'File is not found';
         }
+        if (state === 'ready' && isReady(newStateObj)) {
+          this.recentFilesService.addRecentFile(newStateObj.url);
+        }
```

Comment:
the issue with it here is that the code route was duplicating this functionality. I'm not super happy about having this here since that means that everyone that uses a file resource needs to also add this logic. I feel like it should come along for free by merit of using the file resource

Reply:
ya further to the “source of truth” discussion, I favour there being one place that takes care of everything

Reply:
noted. I have reverted the change. Burcu's pr removed `addRecentFiles` from `updateCodePath` which keeps `addRecentFiles` in one place